contracts/instruments/RibbonThetaVault.sol

```diff
@@ -302,32 +294,37 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         );
         require(balanceOf(msg.sender) >= shares, "Insufficient shares");
 
-        queuedWithdrawShares = queuedWithdrawShares.add(shares);
+        _transfer(msg.sender, address(this), shares);
         scheduledWithdrawals[msg.sender] = shares;
+        queuedWithdrawShares = queuedWithdrawShares.add(shares);
```

Comment:
Reorder the lines so that we increment the queuedWithdrawShares first

```solidity
scheduledWithdrawals[msg.sender] = shares;
queuedWithdrawShares = queuedWithdrawShares.add(shares);
_transfer(msg.sender, address(this), shares);
```

This is a good security practice to have, see [Check Effects Interactions](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html).

Reply:
Great reading thanks!

Does this apply to emitting events as well? For example, is it necessary to emit the `ScheduledWithdrawCompleted` event before completing the withdraw to the user?

Reply:
Yep, good practice to emit events before the external contract interactions as well